sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -37,6 +37,19 @@ import org.apache.spark.sql.sources.{ResolvedDataSource, CreateTableUsingAsSelec
 final class DataFrameWriter private[sql](df: DataFrame) {
 
   /**
+   * Check if the provided dataframe has multiple columns
+   * before writing it to an external source
+   */
+  private val error : Unit = {
+    if (df.schema.fieldNames.length != df.schema.fieldNames.distinct.length) {
```

Comment:
I could not figure out how to check if a logical plan has an output operator... Any guidance would help a lot.